docs/core/diagnostics/metrics-instrumentation.md

```diff
@@ -471,3 +524,86 @@ Press p to pause, r to resume, q to quit.
 
 > [!NOTE]
 > OpenTelemetry refers to tags as 'attributes'. These are two different names for the same functionality.
+
+## Testing custom metrics
+
+Its possible to test any custom metrics you add using <xref:Microsoft.Extensions.Telemetry.Testing.Metering.MetricCollector%601>. This type makes it easy to record the measurements
+from specific instruments and assert the values were correct.
+
+### Testing with Dependency Injection
+
+An example test case for code components that use Dependency Injection and IMeterFactory:
+
+```csharp
+public class MetricTests
+{
+    [Fact]
+    public void SaleIncrementsHatsSoldCounter()
+    {
+        // Arrange
+        var services = CreateServiceProvider();
+        var metrics = services.GetRequiredService<HatCoMetrics>();
+        var meterFactory = services.GetRequiredService<IMeterFactory>();
+        var collector = new MetricCollector<int>(meterFactory, "HatCo.Store", "hatco.store.hats_sold");
+
+        // Act
+        metrics.HatsSold(15);
+
+        // Assert
+        var measurements = collector.GetMeasurementSnapshot();
+        Assert.Equal(1, measurements.Count);
+        Assert.Equal(15, measurements[0].Value);
+    }
+
+    // Setup a new service provider. This example creates the collection explicitly but you might leverage
+    // a host or some other application setup code to do this as well.
+    private static IServiceProvider CreateServiceProvider()
+    {
+        ServiceCollection collection = new ServiceCollection();
```

Comment:
>collection [](http://example.com/codeflow?start=26&length=10)

nit: will be better calling it `servceCollection`.

Reply:
But don't forget the `i`.

Reply:
I fixed it :-)